airflow/providers/oracle/hooks/oracle.py

```diff
@@ -90,6 +134,49 @@ def get_conn(self) -> oracledb.Connection:
         mod = conn.extra_dejson.get('module')
         schema = conn.schema
 
+        # Enable oracledb thick mode if thick_mode is set to True, defaults to False
+        # Parameters take precedence over connection config extra
+        # Defaults to False (use thin mode) if not provided in params or connection config extra
+        if self.thick_mode is None:
+            self.thick_mode = conn.extra_dejson.get('thick_mode', False)
+            if not isinstance(self.thick_mode, bool):
+                raise TypeError(f'thick_mode expected bool, got {type(self.thick_mode).__name__}')
```

Comment:
if defining conn in airflow URI you may end up with string value e.g.`'True'`. best to handle this case.